### PR TITLE
[Hub Generated] Review request for Microsoft.ResourceHealth to add version stable/2022-10-01

### PR DESCRIPTION
**Pull request by abhayohri**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/abhayohri-resourcehealth-Microsoft.ResourceHealth-2022-10-01...main/).
## ARM API Information (Control Plane)

Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by abhayohri** [comment]
image

**Comment by abhayohri** [comment]
Hey this Pr was built to publish the 2022-10-01 swagger api version. this is based off of the 2022-10-01-preview version through the Open api hub.

there were a few linter errors:

image

These were fixed in 99f7003ea349f77c1b10395d4bd81ca472d9bb03 by adding the missing endpoints

next I added suppression for the error response: ece7d828ada4c9e16027936bfff3d5e8c6caf495

lastly we neaded to add properties to existing objects as these are being flagged by s360:
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=260F1CBF-7480-46E2-B8E1-2C5C518FD39A 
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=7A661FA3-A486-491A-96A2-D68649102642
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=D0645A0D-77A7-4AB5-B734-D42B2E3BEBEB
https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=647F3368-7670-4ADF-9D8D-6A6F02006D3F

The properties seem to have been added to the code but never made it to swagger
events endpoints: added to response code on 5/16/2019 for all api version
properties.article.articleId (string) 
properties.article.parameters (IReadOnlyDictionary<string, string>)
properties.reason (string)   

Availability status endpoints:
properties.ArticleId (string)-> added in 1/13/2022 to all api versions
properties.Context (string) -> added in 1/13/2022 to all api versions
properties.Category (string)-> added on 12/8/2022 to all api versions
properties.recommendedActions[*]._ActionUrl.Comment (string)-> added on 11/14/2018 to all api version 

(adding this last property seems to be triggering a linter exception for [DefinitionsPropertiesNamesCamelCase](https://github.com/Azure/azure-openapi-validator/blob/main/docs/definitions-properties-names-camel-case.md) but the name cannot be changed as it is been served to clients for years)

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:884]: >/{resourceUri} [](http://example.com/codeflow?start=5&length=14)

I think `{resourceUri}` should normally be assume to include the initial '/' character, like a resource's `id` property, so I think you shouldn't prefix it with another one.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:893]: >resourceUri [](http://example.com/codeflow?start=21&length=11)

In other APIs like RBAC, I think I've seen this parameter called 'scope', and so its been sort of canonicalized as a common Parameter in common-types json if you want to use it, or copy it more closely.

'Scope', by its name, does kind of imply a scope which might NOT be a resource. E.g. a resource group or subscription. So I'm perfectly fine with you also choosing NOT to reuse it, and to keep calling your parameter ResourceIdParameter or ResourceUri. Maybe we should even consider establishing this as a new pattern extension resources to follow? What do you all think? :)

ScopeParameter v5 today looks like this:
```
    "ScopeParameter": {
      "name": "scope",
      "in": "path",
      "required": true,
      "type": "string",
      "description": "The scope at which the operation is performed.",
      "minLength": 1,
      "maxLength": 90,
      "x-ms-parameter-location": "method",
      "x-ms-skip-url-encoding": true
    },
```

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:897]: >fully qualified ID of the resource, [](http://example.com/codeflow?start=32&length=35)

[in relation to my other comment - fully qualified ID implies it already has a leading slash right?]

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:2303]: These seem like more things we could have a general form of in common-types. Maybe. Just an observation.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:2287]: I recommend not declaring your own ResourceGroupNameParameter and just instead do $ref to common-types one everywhere.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:884]: [Unfortunate if this feedback wasn't provided on earlier api-versions.]

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:920]: >ErrorResponse [](http://example.com/codeflow?start=37&length=13)

*Error responses - use common types*
There are definitions of error responses and details in commontypes types.json that I recommend you use instead of declaring your own.

See e.g. ErrorResponse and ErrorDetails in 

[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v5/types.json#Definitions](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v5/types.json#Definitions)

Comment (TimLovellSmith) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:920]: Same suggestion applies elsewhere.
This suggestion is OPTIONAL and can be ignored if it would be a breaking change.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json:3]: >resourceUri [](http://example.com/codeflow?start=20&length=11)

should be easy to give a realistic example matching the below?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json:19]: Curious question. How accurate is this. Would it actually be returned as empty, or would it just be omitted from the response?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json:52]: Curious question. How accurate is this. Would they actually be returned as empty strings, or would they just be null or omitted from the response?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:1220]: nit Would it be useful to document which fields are localized when translations are available?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatus_GetByResource.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatus_GetByResource.json:22]: >recommendedActions [](http://example.com/codeflow?start=11&length=18)

why isn't this propertyname recommendedActions camelCased?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:1856]: >resourcehealth [](http://example.com/codeflow?start=62&length=14)

nit, should it use the actual provider name 'Microsoft.ResourceHealth'? Or go for fully conversational style? 'the resource health resource provider'?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:1914]: >url [](http://example.com/codeflow?start=94&length=3)

nit URL

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:2113]: >Resource [](http://example.com/codeflow?start=94&length=8)

Why Resource, and not ProxyResource?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [inline specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:1911]: Why is it this funny looking property name '_ActionUrl.Comment'?

Comment (TimLovellSmith) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:1911]: And not e.g. 'actionUrlComment'?

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:920]: yes this would result in a breaking change unfortunately. we have discussed this with breaking change review board and they asked us to not change it and keep it the same for all endpoints

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:1220]: will create a separate task for this.

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:1911]: apparently this is an old field that is still supported. we recieved the following s360: https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=647F3368-7670-4ADF-9D8D-6A6F02006D3F 

we can't change it without a breaking change

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatus_GetByResource.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatus_GetByResource.json:22]: to give context this is an existing endpoint that has apparently been missing from the swagger in later versions. we can't change the name without a breaking change

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/examples/ChildAvailabilityStatuses_List.json:19]: surprisingly yes

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:884]: I think this is minor and should not stop this from pr from going live. However my thoughts on this is that i believe when this was first created it may have been modeled similarly to the SubscriptionIdParameter from the common types:
we use subscriptions/{subscriptionId}/provides and not subscriptions{subscriptionID}providers 
the first is a lot more readable. 
since we give an example in the description this should not be confusing to clients

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:897]: please see above comment I can remove the "fully qualified"from the description but i think this is minor since we give an example

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json]: fixed

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json]: fixed

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json]: fixed

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json
Comment (TimLovellSmith) [reply specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/stable/2022-10-01/ResourceHealth.json:920]: Sounds good. I defer to breaking change review board in this case.

**Comment by visingla-ms** [comment]
SigningOff from ARM side as checked that comments from previous ARM API Reviewer are resolved. There is one Linter issue which needs to be resolved (fixed/suppressed).

**Comment by lirenhe** [comment]
@abhayohri, the config for dotnet sdk is not correct, could you please update it by following https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/samplefiles/readme.csharp.md

@MaryGao  @tadelesh,  could you take a look at the SDK breaking in this PR, thanks

**Comment by weidongxu-microsoft** [comment]
Please follow on Renhe's comment.

I will skip LintDiff on this single error, as suppression seems not able to work, before merge.

**Comment by lirenhe** [comment]
The suppression doesn't seem to work but it should not be a blocker for this PR.

**Comment by MaryGao** [comment]
Approved JS breaking since it is not introduced by this pr.

File: specification/resourcehealth/resource-manager/readme.md
Comment (weshaggard) [inline specification/resourcehealth/resource-manager/readme.md:130]:
```suggestion
    where: $.definitions.recommendedAction.properties["_ActionUrl.Comment"]
```
Shouldn't that property have an underscore in front? Maybe this is why it isn't being honored.

Comment (abhayohri) [reply specification/resourcehealth/resource-manager/readme.md]: yeah i did try the above format. just tried the one without the underscore last.  i've added the underscore back

**Review by lirenhe — APPROVED** [review]

File: specification/resourcehealth/resource-manager/readme.md
Comment (abhayohri) [reply specification/resourcehealth/resource-manager/readme.md]: hmm after making the change i don't see the error pop up as a lint diff. here is the original commit where I used the underscore https://github.com/Azure/azure-rest-api-specs/pull/22721/commits/809c958cf6febd52236c13459b724752f3d82768 

I may have messed up and thought it was still giving me the error. I apologies for raising this issue . Thank you verry much for your help!

Comment (weshaggard) [reply specification/resourcehealth/resource-manager/readme.md]: The original commit was missing `.properties`

Comment (abhayohri) [reply specification/resourcehealth/resource-manager/readme.md]: ah got it thank you again!

**Pull request merged**
